src/app/test-history/test-history.component.html

```diff
@@ -0,0 +1,42 @@
+<table class="govuk-table t-history">
+  <thead class="govuk-table__head">
+  <tr class="govuk-table__row">
+    <th scope="col" class="govuk-table__header">Test type</th>
+    <th scope="col" class="govuk-table__header">Test date</th>
+    <th scope="col" class="govuk-table__header">Expiry date</th>
+    <th scope="col" class="govuk-table__header">Certificate number</th>
+    <th scope="col" class="govuk-table__header">Result</th>
+  </tr>
+  </thead>
+  <tbody class="govuk-table__body" *ngIf="testResultJson; else noTestHistory">
+  <tr class="govuk-table__row" *ngFor="let testRes of testResultJson; let i = index">
+    <td id="{{ 'test-testCode-' + i }}" class="govuk-table__cell">
+      {{ techRecHelpers.isNullOrEmpty(testRes.testTypes[0].testCode) ? "-" : testRes.testTypes[0].testCode.charAt(0).toUpperCase() + testRes.testTypes[0].testCode.substr(1).toLowerCase() }}
+    </td>
+    <td id="{{ 'test-testEndTimestamp-' + i }}" class="govuk-table__cell">
+      {{ techRecHelpers.isNullOrEmpty(testRes.testEndTimestamp) ? "-" : testRes.testEndTimestamp | date :'dd/MM/y'}}
+    </td>
+    <td id="{{ 'test-expiryDate-' + i }}" class="govuk-table__cell">
+      {{ techRecHelpers.isNullOrEmpty(testRes.testExpiryDate) ? "-" : testRes.testExpiryDate | date :'dd/MM/y'}}
+    </td>
+    <td id="{{ 'test-certificateNumber-' + i }}" class="govuk-table__cell">
+      {{ techRecHelpers.isNullOrEmpty(testRes.testTypes[0].certificateNumber) ? "-" : testRes.testTypes[0].certificateNumber }}
+    </td>
+    <td id="{{ 'test-testResult-' + i }}" class="govuk-table__cell govuk-!-font-weight-bold"
+        [ngStyle]="testRes.testTypes[0].testResult=='pass' ? { color:'#00703C' } : testRes.testTypes[0].testResult=='prs' ? {color: '#1D70B8'} : (testRes.testTypes[0].testResult=='fail' || testRes.testTypes[0].testResult=='abandoned') ? {color: '#D4351C'} : {color: ''}">
+      {{ techRecHelpers.isNullOrEmpty(testRes.testTypes[0].testResult) ? "-" : testRes.testTypes[0].testResult | uppercase }}
+    </td>
+  </tr>
+  </tbody>
+  <ng-template #noTestHistory>
+    <tbody class="govuk-table__body">
+    <tr>
+      <td>-</td>
+      <td>-</td>
+      <td>-</td>
+      <td>-</td>
+      <td>-</td>
+    </tr>
+    </tbody>
+  </ng-template>
```

Comment:
why do we need this ?

Reply:
This template is displayed when there are no results in testResultJson

Reply:
The default template should not be a hardcoded row. It should be mapped to the actual testResult object and the initialState of the object would be using the pipe which we can add for the "-".